File: src/cspfj/heuristic/DDegOnDom.java
package cspfj.heuristic;

import cspfj.problem.Problem;
import cspfj.problem.Variable;

public final class DDegOnDom extends AbstractVariableHeuristic {

	public double getScore(final Variable variable) {
        return variable.getDDeg() / variable.getDomainSize();
    }
	
	public String toString() {
		return "max-ddeg/dom";
	}
}

File: src/main/scala/cspfj/heuristic/DDegOnDom.scala
package cspfj.heuristic;

import scala.collection.JavaConversions
import cspfj.constraint.Constraint
import cspfj.problem.Problem
import cspfj.problem.Variable;
import scala.annotation.tailrec

final class DDegOnDom(val problem: Problem) extends VariableHeuristic with RandomBreak {

  def score(variable: Variable) =
    variable.getDDeg.toDouble / variable.dom.size

  override def toString = "max-ddeg/dom"

}
